File: app/src/main/java/echo/com/surveys/fragment/SurveyFragment.java
package echo.com.surveys.fragment;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import echo.com.surveys.databinding.LayoutSurveyFragmentBinding;
import echo.com.surveys.model.SurveyModel;
import echo.com.surveys.util.ImageUtil;

public class SurveyFragment extends BaseFragment {
    protected LayoutSurveyFragmentBinding binding;
    protected SurveyModel survey;

    private static final String KEY_SURVEY = "KEY_SURVEY";
    public static SurveyFragment getInstance(SurveyModel survey){
        SurveyFragment fragment = new SurveyFragment();
        Bundle bundle =new Bundle();
        bundle.putSerializable(KEY_SURVEY, survey);
        fragment.setArguments(bundle);
        return fragment;
    }


    public SurveyFragment() {
    }


    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = LayoutSurveyFragmentBinding.inflate(inflater, container, false);
        handleBundle();
        return binding.getRoot();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        initializeResources();
    }

    protected void initializeResources() {
//        DisplayMetrics displayMetrics = getResources().getDisplayMetrics();
//        float dpHeight = displayMetrics.heightPixels / displayMetrics.density;
//        float dpWidth = displayMetrics.widthPixels / displayMetrics.density;

        binding.setSurvey(survey);
        ImageUtil.loadResizedImage(getActivity(), binding.image, survey.getHDUrl());
    }

    @Override
    protected void handleBundle() {
        survey = (SurveyModel) getArguments().getSerializable(KEY_SURVEY);
    }
}

File: app/src/main/java/echo/com/surveys/fragment/SurveyFragment.kt
package echo.com.surveys.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import echo.com.surveys.databinding.LayoutSurveyFragmentBinding
import echo.com.surveys.model.SurveyModel
import echo.com.surveys.util.ImageUtil


class SurveyFragment : BaseFragment() {
    lateinit var binding: LayoutSurveyFragmentBinding
    lateinit var survey: SurveyModel


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = LayoutSurveyFragmentBinding.inflate(inflater, container, false)
        handleBundle()
        return binding.root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        initializeResources()
    }

    protected fun initializeResources() {
        binding.survey = survey
        ImageUtil.loadResizedImage(activity!!, binding.image, survey.getHDUrl())
    }

    override fun handleBundle() {
        survey = arguments!!.getSerializable(KEY_SURVEY) as SurveyModel
    }

    companion object {

        private val KEY_SURVEY = "KEY_SURVEY"
        fun getInstance(survey: SurveyModel): SurveyFragment {
            val fragment = SurveyFragment()
            val bundle = Bundle()
            bundle.putSerializable(KEY_SURVEY, survey)
            fragment.arguments = bundle
            return fragment
        }
    }
}
